### DISCUSSION TITLE
Retraining pretrained NER models with new data samples

### DISCUSSION THREAD

**Discussion by vlreinier** [JohnSnowLabs/spark-nlp#1278] [Ideas]
**Describe the solution you'd like**

Is it currently possible to retrain a NerDLApproach model with data new samples (instead of retraining complete dataset + new samples)?

Thanks in advance!

**Comment by maziyarpanahi** [comment]
Hi,

Unfortunately, that is not possible at the moment. We don't have any immediate plan for this as it requires lots of testings for proven practicality.

**Comment by vlreinier** [comment]
Ok thanks for your quick response :)

**Comment by maziyarpanahi** [comment]
Since this question comes up every once in a while, to understand the use cases and possibilities, other approaches, and existing features, etc. 

I would like to turn this into a discussion. This allows other users to chime in and continue the conversation so when we do plan it we know what's required and how it will be used.

**Reply by JJorczik** [reply]
Are there any updates on this topic? It would be very helpful to retrain pre-trained models/pipelines in general instead of extending the training dataset and train from scratch.

Thank you for your help!

**Reply by maziyarpanahi** [reply]
There are no updates on this feature. The current NER architecture by using Word Embeddings won't be possible to be used as a few-shot token classification. I have done some experiments, the dataset must have the exact same labels within the exact same domain, or it won't learn anything and may forget what it already has learned. Even then there is no evidence it does learn new data or improve anything.

This can be only possible by those giant language models like GPT and so far I only saw text classification (multi-class) for extending new classes not named entity recognition. (in practice)

**Reply by JJorczik** [reply]
Thank you for your quick response. My question was also more related to models in general and not specifically NER. So retraining a pre-trained multi-class text classification model for including new classes would be applicable in Spark NLP or does it come with similar problems as NER architecture?

**Reply by trkrishnan** [reply]
I have the same question . Can you please tell me if I can retrain any NER pretrained model by adding new entities to it . If NER pretrained model needs to be fully retrained , is there any documents or links that gives the procedure to retrain from scratch by adding new entities to be added to the training data through API or any other procedures

**Reply by maziyarpanahi** [reply]
The answer is still the same. There is no way to continue, retrain, or add any new entities to already train NER models. You need to train a new model from scratch based on your own training dataset. 

We have examples, tutorials, articles, etc. but just in case: https://github.com/JohnSnowLabs/spark-nlp-workshop/blob/master/tutorials/Certification_Trainings/Public/4.NERDL_Training.ipynb

**Reply by trkrishnan** [reply]
Thanks for the quick response
